frontend/src/components/StockData/StockData.jsx

```diff
@@ -61,11 +62,12 @@ export default function StockData(props) {
                 // POST stock data to database for later use
                 axios.post("http://localhost:3000/stocks", combinedStockData);
 
-                setStockNotFound(false);
             } catch (error) {
                 setStockNotFound(true);
             } finally {
-                setIsLoading(false);
+                setTimeout(() => {
+                    setIsLoading(false);
```

Comment:
I think ending the loading animation should be right after you setStockData at line 60. 
Please try it out and let us know if it works.

Reply:
If the stock is already in the database we never get to line 60 so that's why I put it in finally. Let me know if there is a way to change it.

Reply:
OK. Then you can also set Loading to FALSE right after `setStockData(databaseResponse.data);` in line 27.